constraints/binRightmost.js

```diff
@@ -0,0 +1,60 @@
+/* Assign a violation for every node of category cat 
+such that its rightmost child of category (cat-1) 
+has more than two children.
+*/
+
+function binMaxRightmostBranches(s, ptree, cat) {
+  var vcount = 0;
+  //base case: we are at leaf && there are no children
+  //make sure there is children
+  if (ptree.children && ptree.children.length) {
+    if (ptree.cat === cat) {
+      //check rightmost child
+      var rightMost = ptree.children.length - 1;
+      var rightMostChild = ptree.children[rightMost];
+      if (rightMostChild.children && rightMostChild.children.length > 2) {
+        vcount++;
+      }       
+    }
+    //check other nodes in ptree
+    for(var i = 0; i < ptree.children.length; i++) {
+      vcount += binMaxRightmostBranches(s, ptree.children[i], cat);
+    }       
+  }
+  return vcount;
+};
+var parent_ptree;
+function binMaxRightmostLeaves(s, ptree, cat) {
+  //first find the first level of with cat --> recursive look into the children
+  var vcount = 0;
+  if(ptree.children && ptree.children.length) {
+  if(ptree.cat === cat) {
+    var recur_flag = 0;
+    for(var i = 0; i < ptree.children.length; i++) {
+      if(ptree.children[i].cat === cat) {
```

Comment:
Here vcount is being incremented without checking whether there are more than 2 nodes of cat - 1 dominated. For example, let toomanyvtree = {((a))((b))((c))}, where {} stands for iota and () stands for phi, and a, b, c are ws. Then binMaxRightmostLeaves({}, toomanyvtree, "phi") should be 0. But the current implementation incorrectly returns 3 for this case. 
This check has nothing to do with whether there is actually a violation of the constraint.